llvm/test/CodeGen/RISCV/rv64zba.ll

```diff
@@ -2681,7 +2673,7 @@ define i64 @srliw_3_sh3add(ptr %0, i32 signext %1) {
 ; RV64ZBA-LABEL: srliw_3_sh3add:
 ; RV64ZBA:       # %bb.0:
 ; RV64ZBA-NEXT:    srliw a1, a1, 3
-; RV64ZBA-NEXT:    sh3add.uw a0, a1, a0
```

Comment:
srliw always puts 0s in the upper 32 bits so the .uw here isn't necessary. So I don't think the mistake can cause a miscompile.